region/norfair/east.json

```diff
@@ -2930,36 +2938,102 @@
             {
               "id": 1,
               "requires": [
+                {"or": [
+                  "heatProof",
+                  "canHeatRun"
+                ]},
                 {"or": [
                   {"and": [
                     "canSwim",
                     {"or": [
-                      "SpaceJump",
+                      {"and": [
+                        "SpaceJump",
+                        {"cost": [
+                          {"heatFrames": 600}
+                        ]}
+                      ]},
+                      {"and": [
+                        "SpeedBooster",
+                        "canDamageBoost",
+                        {"canComeInCharged":{
+                          "fromNode": 2,
+                          "framesRemaining": 180
+                        }},
+                        {"cost": [
+                          {"heatFrames": 500},
+                          {"shinesparkFrames": 33},
+                          [
+                            {"enemyDamage": {
+                              "enemy": 93,
+                              "hits": 1
+                            }}
+                          ]
+                        ]}
+                      ]},
                       {"and": [
                         "canLavaGravityJump",
                         {"or": [
-                          "HiJump",
-                          "canGravityWalljump",
+                          {"and": [
+                            "HiJump",
+                            {"cost": [
+                              {"heatFrames": 600},
+                              {"lavaFrames": 100}
+                            ]}
+                          ]},
                           {"and": [
                             "canSuitlessLavaDive",
-                            "canSuitlessLavaWalljump"
+                            "canSuitlessLavaWalljump",
+                            {"cost": [
+                              {"heatFrames": 750},
+                              {"lavaFrames": 250}
+                            ]}
                           ]}
                         ]}
                       ]}
                     ]}
                   ]},
                   {"and": [
                     "canSuitlessLavaDive",
-                    "canSuitlessLavaWalljump",
+                    "Morph",
```

Comment:
Feels kinda hard to explain that you absolutely need Morph to do a lava dive, no?